2_segmenting_data/README.md

```diff
@@ -0,0 +1,104 @@
+
+## Cytoplasm Segmentation
+
+After experimenting with the various parameters in CellPose on all of the pilot NF1 DAPI (RFP) channel images, we settled on the following parameters for CellPose cytoplasm segmentation:
+- `model_type : "cyto2"` This parameter sets the model from CellPose to be used as the cytoplasm 2 model, which I found segments cytoplasm better than the other models.
+- `channels : [1,3]` This parameter sets the channels to be used by the model to segment the cytoplasm of the cells using the nuclei as the base (1: blue) and the RFP channel as the channel to be segmented (3: red).
+Since CellPose struggled with segmenting cytoplasm using only the RFP channel, I use a function to overlay nuclei, ER, and cytoplasm channels into one image from each well and site called `overlay_channels()` in [segmentation_utils.py](segmentation_utils.py).
+ER channel is not used by CellPose during segmentation as CellPose can only use 2 channels, one as a base and one to segment.
+
+**Note:** The channel numbers correspond to the CellPose channel colors.
+The `overlay_channels()` function makes cytoplasm the red channel of the image and nuclei the blue channel of the image.
+Thus, `channels : [1,3]` has CellPose segment the cytoplasm channel (3: red) using the nuclei channel (1: blue) as its base.
+
+- `diameter : 155` This parameter sets the cell diameter of the cells within the image to a value, which can be calculated using the `calculate` button. 
+I found that the segmentation worked better when the diameter was established with a set value.
+- `flow_threshold : 0.4` This paramenter decreases the maximum allowed error of the flows for each mask (default is `flow_threshold : 0.4`).
+- `remove_edge_masks : True` This parameter removes any masks from CellPose that are touching an edge of the image.
+
+To show how I determined the best parameters for cytoplasm segmentation, below is a figure showing a cropped image of cells from the cytoplasm channel (**panel A**) and various cytoplasm segmentation methods. 
+
+![Cytoplasm_Segmented](Example_Figures/Cytoplasm_Segmentation_Fig.png)
+
+In **panel B**, it shows how CellPose does with only segmenting cytoplasm using the greyscale image from the RFP channel.
+Since it has no indication of what is a cell, it struggles with any parameters for properly segment cytoplasm, which this image is an example of one set of parameters.
+
+In **panel C**, this shows how CellPose segments the cytoplasm with non-optimal parameters and with the channels overlayed into a single image. 
+As seen with the cell with the orange filter, these parameters oversegments the cell and does not keep the shape of that cell. 
+As well, with the green filtered cell to the left of the orange cell, CellPose undersegments this cell and does not include a lot of the cytoplasm within the outlines. 
+
+In **panel D**, this image shows the parameters I used to segment the cytoplasm.
```

Comment:
This really does look beautiful. Well done getting us to this point!

I want to confirm that you've checked these settings on multiple images. Is this true? Sometimes what works for one image may work terribly for others.

Reply:
I just went through all the images again with those parameters and I wasn't happy with it. I was playing around with it and found the big parameter to be the cell diameter. No matter what, there was always a few images that didn't like those parameters. But, there wasn't a pattern in that one genotype struggled with the parameters more than another. I found that decreasing the cell diameter value a little bit did help but the common problem images are: C7-3, E6-1, E6-3, E6-4, E7-4, and F6-3. 

Will need to rerun the segmentation pipeline as I made changes.

Reply:
Has been changed from 155 to 146 for diameter after our discussion.